mm2src/mm2_main/src/rpc/streaming_activations/swaps.rs

```diff
@@ -0,0 +1,30 @@
+//! RPC activation and deactivation of the swap status streamer.
+use super::{EnableStreamingRequest, EnableStreamingResponse};
+use crate::lp_swap::swap_events::SwapStatusStreamer;
+use mm2_core::mm_ctx::MmArc;
+use mm2_err_handle::{map_to_mm::MapToMmResult, mm_error::MmResult};
+
+use common::HttpStatusCode;
+use http::StatusCode;
+
+#[derive(Display, Serialize, SerializeErrorType)]
+#[serde(tag = "error_type", content = "error_data")]
+pub enum SwapStatusStreamingRequestError {
+    EnableError(String),
+}
+
+impl HttpStatusCode for SwapStatusStreamingRequestError {
+    fn status_code(&self) -> StatusCode { StatusCode::BAD_REQUEST }
```

Comment:
`match self` should be preferred here to ensure we will not miss other variants when added.

Reply:
[547fc2b](https://github.com/KomodoPlatform/komodo-defi-framework/pull/2172/commits/547fc2bb4c03980379cfd219995e1d5ffa199bde)